src/devices/cpu/ht1130/ht1130.h

```diff
@@ -0,0 +1,127 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+
+#ifndef MAME_CPU_HT1130_HT1130_H
+#define MAME_CPU_HT1130_HT1130_H
+
+#pragma once
+
+
+class ht1130_device : public cpu_device
+{
+public:
+	ht1130_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	enum {
+		HT1130_PC,
+		HT1130_R0,
+		HT1130_R1,
+		HT1130_R2,
+		HT1130_R3,
+		HT1130_R4,
+		HT1130_ACC,
+		HT1130_TIMER_EN,
+		HT1130_TIMER,
+	};
+
+	auto pm_in_cb() { return m_port_in_pm.bind(); }
+	auto ps_in_cb() { return m_port_in_ps.bind(); }
+	auto pp_in_cb() { return m_port_in_pp.bind(); }
+
+	auto pa_out_cb() { return m_port_out_pa.bind(); }
+
+	auto display_offset_out_cb() { return m_display_offset_out.bind(); }
+	auto display_data_out_cb() { return m_display_data_out.bind(); }
+
+protected:
+	ht1130_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, u32 clock, address_map_constructor data);
+
+	virtual void device_start() override;
+	virtual void device_reset() override;
+
+	virtual void execute_run() override;
+	virtual void execute_set_input(int inputnum, int state) override;
+
+	virtual std::unique_ptr<util::disasm_interface> create_disassembler() override;
+
+	virtual space_config_vector memory_space_config() const override;
+
+	void internal_map(address_map &map);
+	void internal_data_map(address_map &map);
+
+	inline void tempram_w(offs_t offset, u8 data);
+	inline void displayram_w(offs_t offset, u8 data);
```

Comment:
It’s generally not a great idea to put the `inline` qualifier on member function declarations.  It’s better to put it on the definitions.

Reply:
what's the reference for this? I'll change it, but I get completely opposite advice on work projects